core/src/main/java/org/apache/iceberg/RemoveSnapshots.java

```diff
@@ -78,6 +81,13 @@ public void accept(String file) {
   RemoveSnapshots(TableOperations ops) {
     this.ops = ops;
     this.base = ops.current();
+
+    ValidationException.check(
+        PropertyUtil.propertyAsBoolean(base.properties(), GC_ENABLED, GC_ENABLED_DEFAULT),
+        "Not allowed to expire snapshots as the garbage collection is disabled for this table. " +
```

Comment:
nit 'the' not needed: Not allowed to expire snapshots as garbage collection is disabled for this table.